asv/benchmark.py

```diff
@@ -340,6 +340,15 @@ def get_source_code(items):
     return "\n\n".join(sources).rstrip()
 
 
+def _repr_no_address(obj):
+    result = repr(obj)
+    address_regex = re.compile('^<.* at 0x[\da-fA-F]*>$')
+    if address_regex.match(result):
+        result = '<{} {}>'.format(str(obj.__class__.__name__), str(obj.__name__))
```

Comment:
This needs at least to guard against missing attributes, as `obj` may be any object whatsoever, and the repr may contain anything. I'd actually prefer to guard this with `if isinstance(obj, ...):`, so that the way the transformation is applied is actually controlled.

Reply:
Fair point, I'll see if I can accomplish the same result solely by modifying the `repr()` string, which should resolve your concern.

Reply:
I'm a bit worried about applying modifications to the repr string blindly --- the problem with this is that it's quite obscure behavior, and if something goes wrong it will be difficult to understand...

Reply:
Okay, at the extreme end of options is to just use `IPython.lib.pretty.pprint()`, which handles this on a type-by-type basis (and produces identical output for functions as here). However, that still leaks memory addresses in a bunch of cases and adds a dependency to a sensitive file.

Would the following be good enough in your view?
* If the regex triggers, extract the suspected memory address.
* Then compute `object.__repr__(obj)` and extract a known memory address.
* Iff they match, delete the memory address substring.